addons/sale_project/tests/test_project_project.py

```diff
@@ -29,3 +29,49 @@ def test_projects_to_make_billable(self):
         )[0]
         projects_to_make_billable += non_billable_projects
         self.assertEqual(projects_to_make_billable, project1 + project2)
+
+    def test_default_get_create_project(self):
+        """ Test the default_get method when creating a project """
+        Project = self.env['project.project']
+        partner = self.env['res.partner'].create({'name': 'Test Partner'})
+        sale_order = self.env['sale.order'].create({'name': 'Test Sale Order', 'partner_id': partner.id})
+        service_product_1 = self.env['product.product'].create({'name': 'Service Product 1', 'type': 'service'})
+        service_product_2 = self.env['product.product'].create({'name': 'Service Product 2', 'type': 'service'})
+        self.env['sale.order.line'].create({
+            'order_id': sale_order.id,
+            'product_id': service_product_1.id,
+            'name': 'Test Sale Order Line 1',
+            'product_uom_qty': 1,
+            'price_unit': 100,
+            'is_service': True,
+            'sequence': 10,
+        })
+        sale_line_2 = self.env['sale.order.line'].create({
+            'order_id': sale_order.id,
+            'product_id': service_product_2.id,
+            'name': 'Test Sale Order Line 2',
+            'product_uom_qty': 1,
+            'price_unit': 200,
+            'is_service': True,
+            'sequence': 5,
+        })
+
+        context_with_sale_state = {
+            'default_partner_id': partner.id,
+            'order_state': 'sale',
+            'order_id': sale_order.id,
+        }
+        defaults_with_sale_state = Project.with_context(context_with_sale_state).default_get(['reinvoiced_sale_order_id', 'sale_line_id', 'partner_id'])
+        self.assertEqual(defaults_with_sale_state.get('reinvoiced_sale_order_id'), sale_order.id)
+        self.assertEqual(defaults_with_sale_state.get('sale_line_id'), sale_line_2.id)
+        self.assertEqual(defaults_with_sale_state.get('partner_id'), partner.id)
+
+        context_without_sale_state = {
+            'default_partner_id': partner.id,
+            'order_state': 'draft',
+            'order_id': sale_order.id,
+        }
+        defaults_without_sale_state = Project.with_context(context_without_sale_state).default_get(['reinvoiced_sale_order_id', 'sale_line_id', 'partner_id'])
+        self.assertNotIn('reinvoiced_sale_order_id', defaults_without_sale_state)
+        self.assertNotIn('sale_line_id', defaults_without_sale_state)
+        self.assertEqual(defaults_without_sale_state.get('partner_id'), partner.id)
```

Comment:
Move to the test_sale_project.py file. 
- You can use the existing products and partner.
- Don’t manually set the order state in the context. Use the action_confirm method instead.
- Use Form instead of default_get. 
    For example:
   ` with Form(self.env['project.project'].with_context(...)) as project_form:`